src/runtime/manifest-ast-nodes.ts

```diff
@@ -666,20 +666,20 @@ export interface SchemaReferenceType extends BaseNodeWithRefinement {
 
 export interface SchemaUnionType extends BaseNodeWithRefinement {
   kind: 'schema-union';
-  types: string[];
+  types: SchemaPrimitiveType[];
```

Comment:
I think these should probably actually be SchemaTypes.

Reply:
Yep, sounds good.